### ISSUE TITLE
ConnectionMultiplexer.Connect occasionally throwing ArgumentException

### ISSUE THREAD

**Issue by Flavien** [StackExchange/StackExchange.Redis#148]
I am occasionally getting the following exception when calling ConnectionMultiplexer.Connect in Windows Azure.

```
System.ArgumentException: Item has already been added. Key in dictionary: 'Unspecified/xxx.redis.cache.windows.net:6380'  Key being added: 'Unspecified/xxx.redis.cache.windows.net:6380'
at System.Collections.Hashtable.Insert(Object key, Object nvalue, Boolean add)
at StackExchange.Redis.ConnectionMultiplexer.<ReconfigureAsync>d__25.MoveNext()
--- End of inner exception stack trace ---
at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
at StackExchange.Redis.ConnectionMultiplexer.Connect(String configuration, TextWriter log)
```

**Comment by smakus** [comment]
I am getting it too.  I'm not exactly sure what is causing it.  I have the multiplexer being set in a singleton class, and it is supposed to be threadsafe anyway, so its strange.

**Comment by smakus** [comment]
@Flavien have you deployed version 1.0.394?

**Comment by Flavien** [comment]
No I'm running 1.0.316.0.

**Comment by smakus** [comment]
I published 1.0.394 to a new Azure instance a couple days ago, and I haven't encountered the error since.  I'm not sure it's fixed, but so far so good. @Flavien

**Comment by Flavien** [comment]
Good to know, I will try.

**Comment by bansalnavin** [comment]
Hi,
I am experiencing exactly the same issue. I am using 1.0.0317 version. We have developed a common wrapper component over Redis cache which works absolutely fine in most of the azure services except one. In the one where it is not working, I am getting above error. 

Were you able to identify exact root cause of it? I do not want to straight away upgrade to 1.0.0394 as it will impact other azure services. 

Please let me know your findings.

**Comment by NickCraver** [comment]
It looks like this was fixed between 1.0316 and 1.0394 and isn't a current issue, closing this out to tidy up.

**Comment by nertim** [comment]
Hello, I am starting to see similar behavior Azure recently. We're using 1.0.488 version. Error we're seeing is:
System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> System.ArgumentException: Item has already been added. Key in dictionary: '<host>:6380'  Key being added: '<host>:6380'
   at System.Collections.Hashtable.Insert(Object key, Object nvalue, Boolean add)
   at StackExchange.Redis.ConnectionMultiplexer.<ReconfigureAsync>d__33.MoveNext() in c:\TeamCity\buildAgent\work\3ae0647004edff78\StackExchange.Redis\StackExchange\Redis\ConnectionMultiplexer.cs:line 1420

Any ideas?
